vulcanai/datasets/tabulardataset.py

```diff
@@ -1,604 +0,0 @@
-            merge_on_columns: list of strings
-                Key(s) that specifies which columns to use to uniquely stitch
-                dataset (default None)
-            index_list: list of strings
-                List of columns to make the index of the dataframe
-            na_values: The values to convert to NaN when reading from csv
-            dataset_dict: keyword parameter, value is dataframe or path string
-                pandas dataframe assigned to keyword argument that produces a
-                dictionary variable.
-
-        :Example:
-
-        >>> df_test_one: pd.DataFrame({'A': ['A0', 'A1', 'A2', 'A3'],
-                                   'B': ['B0', 'B1', 'B2', 'B3'],
-                                   'C': ['C0', 'C1', 'C2', 'C3'],
-                                   'D': ['D0', 'D1', 'D2', 'D3']},
-                                   index=[0, 1, 2, 3]),
-
-        >>> df_test_two: pd.DataFrame({'A': ['A4', 'A5', 'A6', 'A7'],
-                                    'B': ['B4', 'B5', 'B6', 'B7'],
-                                    'C': ['C4', 'C5', 'C6', 'C7'],
-                                    'D': ['D4', 'D5', 'D6', 'D7']},
-                                    index=[4, 5, 6, 7])}
-        >>> tab_dataset_var = TabularDataset(merge_on_columns=['A'],
-                            index_list=['A'], df1=df_test_one, df2=df_test_two)
-
-    """
-
-    def __init__(self, label_column=None, merge_on_columns=None,
-                 index_list=None, na_values=None, **dataset_dict):
-        """Creates an instance of TabularDataset."""
-        for dataset_name in dataset_dict:
-            dataset_value = dataset_dict[dataset_name]
-            if isinstance(dataset_value, str):
-                f_path = dataset_value
-                dataset_dict[dataset_name] = pd.read_csv(f_path,
-                                                         na_values=na_values,
-                                                         index_col=index_list)
-            elif not isinstance(dataset_value, pd.DataFrame):
-                raise ValueError("Dataset inputs must be either paths \
-                                 or DataFrame objects")
-
-        if len(dataset_dict) == 1:
-            # TODO: do we want to do anything with this name?
-            self.df = dataset_dict[sorted(dataset_dict)[0]]
-        else:
-            # Not using index list now because we set it before
-            self.df = utils.stitch_datasets(dataset_dict, merge_on_columns,
-                                            index_list)
-            # TODO: check index list doesn't fail if set twice...
-
-        self.label_column = label_column
-
-        self.df = utils.clean_dataframe(self.df)
-
-    def __len__(self):
-        """
-        The total number of samples.
-
-        Returns:
-            The total number of samples: int
-        """
-        return len(self.df.index)
-
-    def __getitem__(self, idx):
```

Comment:
What are we gonna use to replace this? how will we adapt this refactor to place nicely with the data loader of tabular data? I guess we would need some sort of `convert_tabular_to_dataset` function?

Reply:
yeah, I'll write that

Reply:
done in most recent commit